backend/src/apiserver/server/util.go

```diff
@@ -216,9 +216,25 @@ func ValidateExperimentResourceReference(resourceManager *resource.ResourceManag
 	return nil
 }
 
+func validatePipelineSpecAndCheckPipelineVersionReference(resourceManager *resource.ResourceManager, spec *api.PipelineSpec, resourceReferences []*api.ResourceReference) error {
```

Comment:
what do you think about "validatePipeline" instead?
`pipelineSpec` and `pipelineVersionReference` seems like implementation details.

EDIT: in fact, after reading `ValidatePipelineSpec` code, I think it's simpler overall to add the new logic to `ValidatePipelineSpec` and include resource references as another input argument to `ValidatePipelineSpec`.

Reply:
I prefer to rename it to "validatePipeline" instead of including resource references as another input argument to ValidatePipelineSpec because resource references is not part of PipelineSpec.

Reply:
I prefer to rename the method to `validatePipelineSpecAndVersionId`.
If include resource references as another input argument to `ValidatePipelineSpec`, I think it may be kind of confusing because resource references  is not part of PipelineSpec